internal/handlers/common/findandmodify.go

```diff
@@ -196,11 +196,11 @@ func PrepareDocumentForUpsert(docs []*types.Document, params *FindAndModifyParam
 // prepareDocumentForInsert creates an insert document from the parameter.
 // When inserting new document we must check that `_id` is present, so we must extract `_id`
 // from query or generate a new one.
-func prepareDocumentForInsert(params *FindAndModifyParams) (*types.Document, error) {
+func prepareDocumentForInsert(docs []*types.Document, params *FindAndModifyParams) (*types.Document, error) {
```

Comment:
`prepareDocumentForInsert` used to be a function that "made" a document based on incoming params. Now it also receieves a slice of documents that are not really used, but just passed to some other functions... Is there a way we can redesign the code to avoid this param?

Reply:
Actually, most codes with prepare naming here are actually no prepare and they need documents from the DB. See `PrepareDocumentForUpsert` for example. I think it was always updating the document instead of preparing. I'll think on this